src/spatialdata/_core/_elements.py

```diff
@@ -0,0 +1,104 @@
+"""SpatialData elements."""
+from __future__ import annotations
+
+from collections import UserDict
+from collections.abc import Iterable
+from typing import Any
+from warnings import warn
+
+from dask.dataframe.core import DataFrame as DaskDataFrame
+from datatree import DataTree
+from geopandas import GeoDataFrame
+
+from spatialdata._types import Raster_T
+from spatialdata._utils import multiscale_spatial_image_from_data_tree
+from spatialdata.models import (
+    Image2DModel,
+    Image3DModel,
+    Labels2DModel,
+    Labels3DModel,
+    PointsModel,
+    ShapesModel,
+    get_axes_names,
+    get_model,
+)
+
+
+class Elements(UserDict[str, Any]):
```

Comment:
Nitpick, but why a `UserDict` here instead of a regular dict?

Reply:
The guidelines seem to be that one subclasses `dict` to extend the interface and one subclasses `UserDict` to override methods, like `__setitem__()` https://stackoverflow.com/a/7148602/3343783. So I would keep `UserDict`.